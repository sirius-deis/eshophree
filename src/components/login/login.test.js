import { render, screen, fireEvent, act } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Login from './login';

import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        user: {
          email: 'test@test.com',
        },
      }),
  }),
);

describe('Login component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({user: {isLoading: false, error: null}});
    store.dispatch = jest.fn();
  });
  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should fire submit event', async () => {
    render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    );
    const emailInput = screen.getByLabelText('Email *');
    fireEvent.change(emailInput, {
      target: {
        value: 'test@test.com',
      },
    });
    const passwordInput = screen.getByLabelText('Password *');
    fireEvent.change(passwordInput, {
      target: {
        value: 'password',
      },
    });
    const form = screen.getByRole('form', { hidden: true });
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(() => {
      fireEvent.submit(form);
    });
    expect(emailInput.value).toBe('');
    expect(passwordInput.value).toBe('');
  });
});
