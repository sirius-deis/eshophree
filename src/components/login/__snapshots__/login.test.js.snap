// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `
<div>
  <form
    aria-label="form"
    class="sc-gTRrQi jTeBHu"
  >
    <div
      class="sc-bgqQcB kPHljY"
    >
      Log in
    </div>
    <div
      class="sc-dicizt fRkMWc"
    >
      <label
        class="sc-iVCKna bFQppz"
      >
        Email
         *
        <input
          class="sc-gxYJeL bHrHca"
          name="email"
          placeholder="Enter your email"
          required=""
          type="email"
          value=""
        />
      </label>
    </div>
    <div
      class="sc-dicizt fRkMWc"
    >
      <label
        class="sc-iVCKna bFQppz"
      >
        Password
         *
        <input
          class="sc-gxYJeL bHrHca"
          minlength="8"
          name="password"
          placeholder="Enter your password"
          required=""
          type="password"
          value=""
        />
      </label>
    </div>
    <div
      class="sc-crozmw cvagfM"
    >
      <p>
        Don't have account?
         
        <a
          href="/signup"
          style="text-decoration: underline; margin-left: 0.3rem;"
        >
          Sign up
        </a>
      </p>
      <a
        href="/forget"
        style="text-decoration: underline;"
      >
        Forget password?
      </a>
    </div>
    <button
      class="sc-kPTPQs sc-dcrmVg bsVtmJ jmwxNy"
      color="var(--bg-color)"
      type="button"
    >
      Login
    </button>
    <div
      class="sc-cJPUEC hKzfTe"
    >
      <div
        class="sc-jeCNp hcCbso"
      >
        <div
          class="sc-jJTsDX bLCjbi"
          width="35"
        />
        <p
          class="sc-eywOmQ gAUFCq"
        >
          Or 
          login
           with
        </p>
        <div
          class="sc-jJTsDX bLCjbi"
          width="35"
        />
      </div>
      <div
        class="sc-cJPUEC hKzfTe"
      >
        <button
          class="sc-kPTPQs sc-dcrmVg jTTnrr jmwxNy"
          type="button"
        >
          [object Object] Facebook
        </button>
        <button
          class="sc-kPTPQs sc-dcrmVg jTTnrr jmwxNy"
          type="button"
        >
          [object Object] Google
        </button>
      </div>
    </div>
  </form>
</div>
`;
